provider/infoblox/infoblox.go

```diff
@@ -157,7 +158,24 @@ func (p *InfobloxProvider) Records(ctx context.Context) (endpoints []*endpoint.E
 			return nil, fmt.Errorf("could not fetch A records from zone '%s': %s", zone.Fqdn, err)
 		}
 		for _, res := range resA {
-			endpoints = append(endpoints, endpoint.NewEndpoint(res.Name, endpoint.RecordTypeA, res.Ipv4Addr))
+			newEndpoint := endpoint.NewEndpoint(res.Name, endpoint.RecordTypeA, res.Ipv4Addr)
+			// Check if endpoint already exists and add to existing endpoint if it does
+			foundExisting := false
+			for _, ep := range endpoints {
+				if ep.DNSName == newEndpoint.DNSName && ep.RecordType == newEndpoint.RecordType {
+					logrus.Debugf("Adding target '%s' to existing A record '%s'", newEndpoint.Targets[0], ep.DNSName)
+					ep.Targets = append(ep.Targets, newEndpoint.Targets[0])
+					foundExisting = true
+					break
```

Comment:
@sagor999 Can we add a test that covers those lines as well?

Reply:
@Raffo done. and added some additional coverage for other areas as well. Let me know what you think.